python/pyspark/ml/param/__init__.py

```diff
@@ -49,11 +53,21 @@ def _copy_new_parent(self, parent):
         else:
             raise ValueError("Cannot copy from non-dummy parent %s." % parent)
 
+    def _validate(self, value):
+        if not self.isValid(value):
+            raise ValueError("{parent} parameter {name} given invalid value {value}"
+                             .format(parent=self.parent, name=self.name, value=str(value)))
+
+    def _convertAndValidate(self, value):
```

Comment:
While this is an internal function, it might still be useful to have some pydoc text for the expected behaviour for ourselves.